src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -69,12 +72,28 @@ public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecu
         break;
       case "MESSAGE":
         // Sends request to execution module
-        String email = event.at("/message/sender/email").asText();
         String message = event.at("/message/text").asText();
         String threadId = event.at("/message/thread/name").asText();
-        replyText = macroExecutionModule.execute(email, message, threadId);
+
+        String[] words = message.split(" "); 
+
+        if (words[1].equalsIgnoreCase("initiate")) {
+            String creator = event.at("/message/sender/email").asText();
+            roomToCreator.put(event.at("/space/name").asText(), creator);
+            replyText = "Your macros have now been initiated for this room. All users in this room can use your macros.";
+        }
+        else {
+            if (roomToCreator.containsKey(event.at("/space/name").asText())) {
```

Comment:
OK, let's assume that the creator called initiate properly before, how do we know that this is the same macro that is being called here. I think the map should be keyed on the room, macro name pair. You can combine them into a single string.